### PR TITLE
Enable Error Prone

### PR DESCRIPTION
**Pull request by al3xliu**
It is easier to restart a new pr than solving the conflicts. (See #42 for previous attempt.)

Also, I think i turn all warnings into error by using "-Werror" option.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -148,6 +152,29 @@ afterEvaluate {
...
+				"-Xlint:-options,-fallthrough",
+				"-Xlint:all",
+		]
+		options.encoding = 'UTF-8'
+		options.fork = true
+		if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
+			options.forkOptions.jvmArgs += ["-Xbootclasspath/p:${configurations.javacJar.asPath}".toString()]
+			options.errorprone.enabled = false
+		} else {
+			options.errorprone.enabled = true
```
Comment (wmdietl) [inline build.gradle:175]:
```suggestion
			options.errorprone.enabled = true
			options.forkOptions.jvmArgs += [
					'--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED',
					'--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED',
					'--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED',
					'--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED',
			]
```

**Review by wmdietl — COMMENTED** [review]
Thanks! Can you try whether my suggested edits work and are formatted properly?

File: build.gradle (hunk 1 above)
Comment (al3xliu) [reply build.gradle:175]: Thanks! Added.

File: build.gradle [hunk 2]
```diff
@@ -148,6 +152,29 @@ afterEvaluate {
...
+	// https://github.com/eisop/checker-framework/blob/567084f11278273b33289d138d9d2a1b513d3c35/build.gradle#L226
+	tasks.withType(JavaCompile) { compilationTask ->
+		sourceCompatibility = 8
+		targetCompatibility = 8
+		options.failOnError = true
+		options.deprecation = true
+		options.compilerArgs += [
+				'-g',
+				'-Werror',
+				"-Xlint:-options,-fallthrough",
```
[the remark this answers is not in the document]
Comment (al3xliu) [reply build.gradle:166]: Thanks!

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**